sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1987,3 +1989,98 @@ case class GpuStringInstr(str: Expression, substr: Expression)
     }
   }
 }
+
+
+
+class GpuConvMeta(
+  expr: Conv,
+  conf: RapidsConf,
+  parent: Option[RapidsMeta[_,_,_]],
+  rule: DataFromReplacementRule) extends TernaryExprMeta(expr, conf, parent, rule) {
+
+  override def tagExprForGpu(): Unit = {
+    val fromBaseLit = GpuOverrides.extractLit(expr.fromBaseExpr)
+    val toBaseLit = GpuOverrides.extractLit(expr.toBaseExpr)
+    (fromBaseLit, toBaseLit) match {
+      case (Some(Literal(fromBaseVal, IntegerType)), Some(Literal(toBaseVal, IntegerType)))
+        if Set(fromBaseVal, toBaseVal).subsetOf(Set(10, 16)) => ()
+      case _ =>
+        willNotWorkOnGpu(because = "only literal 10 or 16 for from_base and to_base are supported")
+    }
+    if (SQLConf.get.ansiEnabled) {
+      willNotWorkOnGpu(because = " the GPU has no overflow checking.")
```

Comment:
I'm not sure we can enable this by default. Even in 3.1.x Spark checks for overflow when encoding the value, and will return -1 if it sees an overflow.  We do not do that.

https://github.com/apache/spark/blob/61e034807dc555d7ceadc534fcb0c82d50fc8719/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/NumberConverter.scala#L56-L59

The ANSI mode fix in 3.4.0 only added in an exception instead of returning -1.

The following was run on Spark 3.3.0
```
scala> val df = Seq("9223372036854775807", "-9223372036854775808", "9223372036854775808", "-9223372036854775809", "10223372036854775807", "-10223372036854775808", null).toDF
scala> df.repartition(1).selectExpr("conv(value, 10, 16)", "value").show(false)
+-------------------+---------------------+
|conv(value, 10, 16)|value                |
+-------------------+---------------------+
|7FFFFFFFFFFFFFFF   |9223372036854775807  |
|8000000000000000   |-9223372036854775808 |
|8000000000000000   |9223372036854775808  |
|7FFFFFFFFFFFFFFF   |-9223372036854775809 |
|8DE0B6B3A763FFFF   |10223372036854775807 |
|721F494C589C0000   |-10223372036854775808|
|null               |null                 |
+-------------------+---------------------+
scala> spark.conf.set("spark.rapids.sql.enabled", false)
scala> df.repartition(1).selectExpr("conv(value, 10, 16)", "value").show(false)
+-------------------+---------------------+
|conv(value, 10, 16)|value                |
+-------------------+---------------------+
|7FFFFFFFFFFFFFFF   |9223372036854775807  |
|FFFFFFFFFFFFFFFF   |-9223372036854775808 |
|8000000000000000   |9223372036854775808  |
|FFFFFFFFFFFFFFFF   |-9223372036854775809 |
|8DE0B6B3A763FFFF   |10223372036854775807 |
|FFFFFFFFFFFFFFFF   |-10223372036854775808|
|null               |null                 |
+-------------------+---------------------+
```

Reply:
I am aware of the -1-equivalent output, i.e conversion to `18446744073709551615` representation as a string in to_base if to_base is positive and `-1` if to_base is negative (which we already fallback to CPU).

```
>>> spark.createDataFrame([('-1',), ('18446744073709551615',), ('18446744073709551616',)], 'a string').selectExpr('conv(a, 10, 10)').show()
+--------------------+
|     conv(a, 10, 10)|
+--------------------+
|18446744073709551615|
|18446744073709551615|
|18446744073709551615|
+--------------------+
```

My reasoning is that since Spark does not allow distinguishing 18446744073709551615 as a result of the overflow check  or based on the original data, it does not really matter. However, it's true that the customer may have a process in place making sure that 18446744073709551615 is uniquely due to an overflow, and filter out these by filtering the output `!= '18446744073709551615'`.

So we can disable by default for unlimited StringType and safely enable of StringType(length) for lengths guaranteed not to overflow.

Reply:
I agree that is probably the reason that Spark added in an ANSI mode because it really is ambiguous. Seeing `FFFFFFFFFFFFFFFF` on the output when converting to hex is ambiguous and you just don't know if an overflow happened or not without further processing.

For me I don't really want to enable this by default if it is only partially done. But I can see an argument for allowing it. @sameerz @mattf do you have an opinion if we should put this in with `conv` enabled by default while we work on a better long term solution? Would it be okay to put it in with `conv` disabled by default and some docs so users know how to enable it if there are potential incompatibilities?